src/tpg_tcp.c

```diff
@@ -643,6 +652,8 @@ static struct tcp_hdr *tcp_build_hdr(tcp_control_block_t *tcb,
     tcp_hdr->tcp_flags = flags & TCP_BUILD_FLAG_MASK;
     tcp_hdr->rx_win = rte_cpu_to_be_16(tcb->tcb_rcv.wnd);
     tcp_hdr->tcp_urp = rte_cpu_to_be_16(0); /* TODO: set correctly if urgen flag is set */
+    mbuf->l4_len = tcp_hdr_len;
+    ip_hdr_len = ((ipv4_hdr->version_ihl & 0x0F) << 2);
```

Comment:
No need to compute `ip_hdr_len` here. It's only used inside the `if` below so I would keep it there.